dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/JsonGenerator.java

```diff
@@ -59,15 +70,81 @@
 
         boolean newLineBeforeArrayEnd;
 
-        String memberIndent;
+        /**
+         * 1 level of indent
+         */
+        String itemIndent;
+    }
+
+    @Value
+    @Builder( toBuilder = true )
+    static class Language
+    {
+
+        public static final Language JSON = new Language(
+            "[", "", "]",
+            "{", ":", "}", ",",
+            "\"", "\"", JsonGenerator::escapeJsonString,
+            "\"", "\"", false, JsonGenerator::escapeJsonText,
+            UnaryOperator.identity() );
+
+        public static final Language YAML = new Language(
+            "", "- ", "",
+            "", ": ", "", "",
+            "", "", JsonGenerator::escapeYamlString,
+            "|-", "", true, JsonGenerator::escapeYamlText,
+            format -> format.toBuilder()
+                .itemIndent( "  " )
+                .newLineBeforeItem( true )
+                .newLineAfterArrayStart( false )
+                .newLineBeforeArrayEnd( false )
+                .newLineAfterObjectStart( false )
+                .newLineBeforeObjectEnd( false )
+                .build() );
```

Comment:
It is possible but not advisable. An intermediate representation was avoided specifically because of size/performance and the additional complexity. This generates YAML just fine. Involve jackson brings extra complexity and costs. Results might change based on version. Corner cases might not do what you want.

Reply:
The main point is that we do not want to create a in-memory object tree representation for performance reasons 
and to the extend they being used here JSON and YAML are quite simple to generate. 

You are right that any dependency has that problem but no dependency does not so that is why I don't use one. Also going to a intermediate object graph hides subtle distinctions. For example, now we found that it makes sense to handle multi-line strings different from "simple" strings. This would have been lost in an object graph. And a transformer would have to work hard to go back to making some strings into a multi-line notation while keeping others in single line notation. Also which of the many multi-line notations would the transformer pick? 

I can't understand why you so desperately want to use the jackson hammer on this? The result is the same so why do you care as a consumer?